pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -171,6 +171,21 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 
 //nolint:gocyclo
 func (c *Controller) finalize(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
+	// perform a no-op for whatever the status condition is currently set to
+	// so that we bump the observed generation to the latest and prevent the nodeclaim
```

Comment:
Can we add a note that the generation bumps here because of the deletionTimestamp being set?